text/0000-rust-governance.md

```diff
@@ -0,0 +1,728 @@
+  (Thanks to Nick Cameron for this suggestion.)
+
+Subteams make it possible to involve a larger, more diverse group in the
+decision-making process. In particular, **they should involve a mix of**:
+
+* Rust project leadership, in the form of at least one core team member (the
+  leader of the subteam).
+
+* Area experts: people who have a lot of interest and expertise in the subteam
+  area, but who may be far less engaged with other areas of the project.
+
+* Stakeholders: people who are strongly affected by decisions in the
+  subteam area, but who may not be experts in the design or
+  implementation of that area. *It is crucial that some people heavily
+  using Rust for applications/libraries have a seat at the table, to
+  make sure we are actually addressing real-world needs.*
+
+Members should have demonstrated a good sense for design and dealing with
+tradeoffs, an ability to work within a framework of consensus, and of course
+sufficient knowledge about or experience with the subteam area. Leaders should
+in addition have demonstrated exceptional communication, design, and people
+skills. They must be able to work with a diverse group of people and help lead
+it toward consensus and execution.
+
+Each subteam is led by a member of the core team. **The leader is responsible for**:
+
+* Setting up the subteam:
+
+    * Deciding on the initial membership of the subteam (in consultation with
+      the core team). Once the subteam is up and running.
+
+    * Working with subteam members to determine and publish subteam policies and
+      mechanics, including the way that subteam members join or leave the team
+      (which should be based on subteam consensus).
+
+* Communicating core team vision downward to the subteam.
+
+* Alerting the core team to subteam RFCs that need global, cross-cutting
+  attention, and to RFCs that have entered the "final comment period" (see below).
+
+* Ensuring that RFCs and PRs are progressing at a reasonable rate, re-assigning
+  shepherds/reviewers as needed.
+
+* Making final decisions in cases of contentious RFCs that are unable to reach
+  consensus otherwise (should be rare).
+
+The way that subteams communicate internally and externally is left to each
+subteam to decide, but:
+
+* Technical discussion should take place as much as possible on public forums,
+  ideally on RFC/PR threads and tagged discuss posts.
+
+* Each subteam will have a dedicated
+  [discuss forum](http://internals.rust-lang.org/) tag.
+
+* Subteams should actively seek out discussion and input from stakeholders who
+  are not members of the team.
+
+* Subteams should have some kind of regular meeting or other way of making
+  decisions. The content of this meeting should be summarized with the rationale
+  for each decision -- and, as explained below, decisions should generally be
+  about weighting a set of already-known tradeoffs, not discussing or
+  discovering new rationale.
```

Comment:
@nrc 

> I think there is some happy place between summaries and literal transcripts that is more ideal than both. I often find detailed minutes extremely useful. Specifically it is good to know who is involved in a discussion and how (so you know who to contact if you want to be part of the conversation), how much confidence there was in the arguments, how close the decision was, what the motivations are, the reasoning that people use. Summaries rarely capture the level of detail necessary for someone not at the meeting to really understand *how* a decision is made.

This puts much more weight on meetings/final point of making a decision than I think it merits.

We've been pushing for the last several months for strong, active participation on RFC threads prior to any decision. The decision-making process at this point is usually just a check: "Has everyone's objections been raised on the thread and responded to"? And the more controversial a topic, the more we strive to debate it vigorously on the RFC thread.

When RFC decisions were previously made in the minuted weekly meeting, there were many problematic cases that resulted in a lot of frustration in the community, despite those minutes. 

I think that the situation has improved greatly over the last several months, mostly due to the increased discipline in the core team spending time engaging directly on RFC threads, laying out arguments, tradeoffs, and worries. These activities all support the goals you mentioned above, but they do so *before* the point of a final decision, and I think that is key.

Whatever venue a decision is made within, I think what people want is a chance to engage with the rationale, and thereby to influence the decision-makers, prior to the final decision. That's not really about meetings or minutes; it's about team members participating actively in all phases of the RFC process.

> I do agree there is a place for private conversation, but it should be much rarer then it currently is. It has its place to do very early brainstorming and later very detailed work, but we would benefit from more of the middle section of idea development being public by default.

I think everyone agrees that conversation should be public by default, and the RFC explicitly mentions this (as did my bullets above).

FWIW, I don't share the impression that much of significance is being discussed privately at the moment. I tried to express this above with the strong move toward participation on RFC threads, which I think has been very successful. It also applies to topics beyond RFC decisions, where anything actionable that comes out of a core team meeting has a next step of getting feedback from the broader community, be it via a discourse thread, a new RFC, or something else.

But in any case, as I've said elsewhere, we can certainly minute the core team meetings (in the traditional sense) going forward.

> In particular, the idea of forming an idea in private and then communicating it publicly seems the wrong approach. The idea should be formed in public (where by public I don't mean the whole world or every Rust user, it might mean just one subteam's peers) and there should be no need to communicate things because the conversation itself is public.

This may just be a difference in working style. I appreciate being able to bounce a passing thought off a collaborator without starting an entire thread or group IRC discussion about it. If it turns into something interesting, then I get feedback from a larger group until I'm ready to write an actual RFC about it.